tools/Azure.Mcp.Tools.ApplicationInsights/src/Commands/AppTrace/AppTraceListCommand.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.Net;
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Commands.Subscription;
+using Azure.Mcp.Core.Extensions;
+using Azure.Mcp.Core.Models.Command;
+using Azure.Mcp.Core.Models.Option;
+using Azure.Mcp.Tools.ApplicationInsights.Models;
+using Azure.Mcp.Tools.ApplicationInsights.Options;
+using Azure.Mcp.Tools.ApplicationInsights.Services;
+using Microsoft.Extensions.Logging;
+using static Azure.Mcp.Tools.ApplicationInsights.Options.ApplicationInsightsOptionDefinitions;
+
+namespace Azure.Mcp.Tools.ApplicationInsights.Commands.AppTrace;
+
+/// <summary>
+/// Command to list Application Insights trace metadata for components in a subscription or resource group.
+/// </summary>
+public sealed class AppTraceListCommand(ILogger<AppTraceListCommand> logger) : SubscriptionCommand<AppTraceListOptions>()
+{
+    private const string CommandTitle = "List Application Insights Trace Metadata";
+    private readonly ILogger<AppTraceListCommand> _logger = logger;
+
+    public override string Name => "list";
+
+    public override string Description =>
+        $$"""
+        List the most relevant traces from an Application Insights table.
+
+        This tool is useful for correlating errors and dependencies to specific transactions in an application.
+
+        Returns a list of traceIds and spanIds that can be further explored for each operation.
+
+        Example usage:
+        Filter to dependency failures
+        "table": "dependencies",
+        "filters": ["success=\"false\""]
+
+        Filter to request failures with 500 code
+        "table": "requests",
+        "filters": ["success=\"false\"", "resultCode=\"500\""]
+
+        Filter to requests slower than 95th percentile (use start and end time filters to filter to the duration spike). Any percentile is valid (e.g. 99p is also valid)
+        "table": "requests",
+        "filters": ["duration=\"95p\""],
+        "start-time":"start of spike (ISO date)",
+        "end-time":"end of spike (ISO date)"
+
+        Use this tool for investigating issues with Application Insights resources.
+        Required options:
+        - {{ResourceName.Name}}: {{ResourceName.Description}} or {{ResourceId.Name}}: {{ResourceId.Description}}
+        - {{Table.Name}}: {{Table.Description}}
+        Optional options:
+        - {{Filters.Name}}: {{Filters.Description}}
+        - {{OptionDefinitions.Common.ResourceGroup.Name}}: {{OptionDefinitions.Common.ResourceGroup.Description}}
+        - {{StartTime.Name}}: {{StartTime.Description}}
+        - {{EndTime.Name}}: {{EndTime.Description}}
```

Comment:
Desc has to be <1024.  And please don't include option info as that is already returned in the tool request